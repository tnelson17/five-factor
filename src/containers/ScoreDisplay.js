import React from 'react';
import Score from '../components/Score.js';
import styled from 'styled-components'

const StyledScore = styled.div`
  color: #0000CD;
`


class ScoreDisplay extends React.Component {
  constructor(){
    super()
    this.state = {
      score: {}
    }
  }


  render(){
    let maxDomainScore = (this.props.testLength / 5) * 4
    return(
      <StyledScore>
        <Score score={this.props.score.O} domain={"Openness to Experience"} max={maxDomainScore}/>
        <Score score={this.props.score.C} domain={"Conscientiousness"} max={maxDomainScore}/>
        <Score score={this.props.score.E} domain={"Extroversion"} max={maxDomainScore}/>
        <Score score={this.props.score.A} domain={"Agreeableness"} max={maxDomainScore}/>
        <Score score={this.props.score.N} domain={"Neuroticism"} max={maxDomainScore}/>
      </StyledScore>
    )
  }
}

export default ScoreDisplay;
