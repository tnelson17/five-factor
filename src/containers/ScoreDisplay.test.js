import React from 'react';
import ScoreDisplay from './ScoreDisplay';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';

const fakeProps = {
  'score': {
    O: 0,
    C: 0,
    E: 0,
    A: 0,
    N: 0
  },
  'testLength': 30
}

it('renders without crashing', () => {
  shallow(
    <ScoreDisplay
      score={fakeProps.score}
      testLength={fakeProps.testLength}
    />
  )
})

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ScoreDisplay
        score={fakeProps.score}
        testLength={fakeProps.testLength}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
