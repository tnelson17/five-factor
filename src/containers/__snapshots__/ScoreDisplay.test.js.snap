// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa bgqgPx"
>
  <div>
    Your score in the dimension of 
    Openness to Experience
     is 
    0
     out of 
    24
  </div>
  <div>
    You are very rarely playful, curious, imaginative, creative,
    open-minded, novelty-seeking, forward-looking
  </div>
  <div>
    Your score in the dimension of 
    Conscientiousness
     is 
    0
     out of 
    24
  </div>
  <div>
    You are very rarely orderly, committed, confident
    (with respect to a sense of mastery), achievement-oriented, reliable, self-disciplined
  </div>
  <div>
    Your score in the dimension of 
    Extroversion
     is 
    0
     out of 
    24
  </div>
  <div>
    You are very rarely active, energetic, thrill-seeking,
    enthusiastic, assertive, interactive, friendly.
  </div>
  <div>
    Your score in the dimension of 
    Agreeableness
     is 
    0
     out of 
    24
  </div>
  <div>
    You are very rarely accommodating, loyal/trusting, compassionate,
    altruistic, steady, cooperative, forgiving/tolerant.
  </div>
  <div>
    Your score in the dimension of 
    Neuroticism
     is 
    0
     out of 
    24
  </div>
  <div>
    You are extremely calm, even-tempered, positive,
    resilient/robust, deliberate, easy-going, regulated.
  </div>
</div>
`;
